File: java/AnoBissexto.java
/**
 * Programa Java define uma função para verificar se um determinado ano é ou não Bissexto.
 *
 * Objetivos:
 *   - uso de funções;
 *   - uso de expressões booleanas (c/ if);
 *   - uso de repetições com o comando for;
 *
 * Módulo: Lógica de programação | Capítulo: XXX | Página: XXX
 * Author: Eder Magalhães
 */
public class AnoBissexto {

    public static void main(String args[]) {
        //verifica um range de anos, a partir de 2005 até 2015
        for (int anoInformado = 2005; anoInformado <= 2015; anoInformado++){
            boolean resultado = ehBissexto(anoInformado);
            System.out.println("Ano "+anoInformado+" é bissexto? "+resultado);
        }
    }

    /*
     * Função recebe como argumento o ano, e retorna boolean indicando se ou nao bissexto.
     */
    static boolean ehBissexto(int ano) {
        return ano % 4 == 0 && ano % 100 != 0 || ano % 400 == 0;
        
	//o trecho a seguir (comentado) é uma outra solução para verificar bissexto, com ifs:

        /*
        if (ano % 4 == 0 && ano % 100 !=0) {
            return true;
        } else if (ano % 400 == 0) {
            return true;
        } else {
            return false;
        }
        */
    }

}

File: python/anoBissexto.py
# -*- coding: utf-8 -*-

# Programa Python define uma função para verificar se um determinado ano é ou não Bissexto.
# Objetivos:
#   - uso de funções;
#   - uso de repetições com o comando for in.
#
# Acesse e execute esse código Python em: http://pythonfiddle.com/ano-bissexto-ap/
#
# Módulo: Lógica de programação
# Author: Eder Magalhães

#Função recebe o argumento ano e retorna o valor booleano indicando se é ou não bissexto.
def ehBissexto(ano):
    return ano % 4 == 0 and ano % 100 != 0 or ano % 400 == 0;

#entrada do programa, faz o laço considerando os anos de 2005 até 2015
for anoInformado in range(2005, 2015):
    resultado = ehBissexto(anoInformado);
    print "Ano ",anoInformado," é bissexto? ",resultado;

